### ISSUE TITLE
[package] luajit is broken because the tarball from the homepage is removed

### ISSUE THREAD

**Issue by uyha** [conan-io/conan-center-index#25032]
### Description

The tarball from https://github.com/conan-io/conan-center-index/blob/master/recipes/luajit/all/conandata.yml is not longer valid since the tarball is removed from the website already.

### Package and Environment Details

.

### Conan profile

.

### Steps to reproduce

.

### Logs

.

**Comment by uilianries** [comment]
I'll open a PR adding mirrors to conandata.yml asap.

**Comment by uilianries** [comment]
After checking the https://luajit.org/download.html they not only removed the download links, but also recommend to not use those old links as well:

> No Release Tarballs or Binaries
>
> LuaJIT uses rolling releases. There are no release tarballs available for download.
> 
> Please do not use obsolete versions from older tarballs or zip files. Please remove any outdated links to these downloads — these links will cease to work soon.
> 
> Do not use pseudo-releases or tarballs created by third parties. Do not use binaries offered for download by third parties. 

After checking the mirror in Github, it does not have the same checksum that old one in luajit website:

    sha256sum 409f7fe570d3c16558e594421c47bdd130238323c9d6fd6c83dedd2aaeb082a8  v2.1.0-beta3.tar.gz

Doing a quick diff with the cached luajit tar ball available in Conan Center, we have:

```
$ diff LuaJIT-2.1.0-beta3-github/ LuaJIT-2.1.0-beta3-jfrog/ -r                                                                                                                                                                                                    
Only in LuaJIT-2.1.0-beta3-github: .gitignore
Only in LuaJIT-2.1.0-beta3-github/src: .gitignore
Only in LuaJIT-2.1.0-beta3-github/src/host: .gitignore
Only in LuaJIT-2.1.0-beta3-github/src/jit: .gitignore
Only in LuaJIT-2.1.0-beta3-jfrog/src: lj_bcdef.h
Only in LuaJIT-2.1.0-beta3-jfrog/src: lj_ffdef.h
Only in LuaJIT-2.1.0-beta3-jfrog/src: lj_folddef.h
Only in LuaJIT-2.1.0-beta3-jfrog/src: lj_libdef.h
Only in LuaJIT-2.1.0-beta3-jfrog/src: lj_recdef.h
```

Those missing headers were auto generated and packaged altogether, but was only available on Luajit download page. You can download the cached one from JFrog here: https://c3i.jfrog.io/artifactory/conan-center-backup-sources/1ad2e34b111c802f9d0cdf019e986909123237a28c746b21295b63c9e785d9c3 (it's a .tar.gz file)

----

The LuaJIT now recommends consuming only rolling releases, but not those old links. Therefore, only replacing the links and checksum value to consume from Github would be wrong as well, as it does not follow the upstream recommendation.

Instead, it would be better updating the recipe, making available the version 2.0 and 2.1, both rolling, and adding an identifier (date, commit ?) to avoid collision and be able to track changes.

There are only other 2 packages in CCI using LuaJIT as requirement and will be affected. Their recipes should be updated after fixing LuaJIT situation:

- libucl/all/conanfile.py:          self.requires("luajit/2.0.5")
- sol2/all/conanfile.py:            self.requires("luajit/2.1.0-beta3")

**Comment by jcar87** [comment]
```
No Release Tarballs or Binaries
LuaJIT uses rolling releases. There are no release tarballs available for download.

Please do not use obsolete versions from older tarballs or zip files. Please remove any outdated links to these downloads — these links will cease to work soon.

Do not use pseudo-releases or tarballs created by third parties. Do not use binaries offered for download by third parties.

Pre-built packages should only be installed via a trusted package manager for your operating system (distro). But you should be aware these often carry old versions that miss important fixes. Before reporting an issue, always try the latest version available from the git repository.

Distro maintainers for distros that require the fiction of a release should do frequent snapshots of a branch. Do not attempt to cherry-pick or backport individual changes, no matter how self-standing individual changes look (because they often are not).
```

This may make this library ineligible for Conan Center altogether.
It's unfortunate - I think tarballs vs rolling are orthogonal concepts. I think we can sympathise with maintainers who are asked to provide support for older releases - but this is a potentially disruptive move. 

Hosting sources is not a Conan Center's goal - we only do a backup for traceability.
